pallets/parachain-staking/src/lib.rs

```diff
@@ -1160,6 +1143,44 @@ pub mod pallet {
     }
 
     impl<T: Config> Pallet<T> {
+        pub fn start_new_era(
+            block_number: T::BlockNumber,
+            mut era: EraInfo<T::BlockNumber>,
+        ) -> (EraInfo<T::BlockNumber>, Weight) {
+            // mutate era
+            era.update(block_number);
+
+            // notify new era has begun
+            let mut weight = T::OnNewEra::on_new_era(era.current);
```

Comment:
the original code was adding this to the weight of base_on_initialize, where the new code with the function discards it.
Is this intended?

Reply:
No it wasn't. Updated to retain the original weight.